url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_AgeGenderLineChart.java

```diff
@@ -0,0 +1,870 @@
+            // Draw median line
+            abLine2D = new Line2D.Double(
+                    lastMeanPointCol, lastPointRow, meanPointCol, pointRow);
+            setPaint(Color.DARK_GRAY);
+            draw(abLine2D);
+            // Set parameters
+            lastMeanPointCol = meanPointCol;
+            lastMeanAddStdDevPointCol = meanAddStdDevPointCol;
+            lastMeanSubtractStdDevPointCol = meanSubtractStdDevPointCol;
+            lastPointRow = pointRow;
+        }
+        /*
+         * Draw Male Lines
+         */
+        firstPoint = true;
+        ite = maleSummaryStatisticsData.entrySet().iterator();
+        while (ite.hasNext()) {
+            entry = ite.next();
+            age = entry.getKey();
+            stats = entry.getValue();
+            firstOrderStats = (BigDecimal[]) stats[0];
+            secondOrderStats = (BigDecimal[]) stats[1];
+            mean = firstOrderStats[1];
+            meanAddStdDev = mean.add(secondOrderStats[5]);
+            meanSubtractStdDev = mean.subtract(secondOrderStats[5]);
+            int meanPointCol = coordinateToScreenCol(mean.negate());
+            int pointRow = coordinateToScreenRow(
+                    BigDecimal.valueOf(age - ageInterval / 2));
+            int meanAddStdDevPointCol = coordinateToScreenCol(
+                    meanAddStdDev.negate());
+            int meanSubtractStdDevPointCol = coordinateToScreenCol(
+                    meanSubtractStdDev.negate());
+            if (firstPoint) {
+                lastMeanPointCol = meanPointCol;
+                lastMeanAddStdDevPointCol = meanAddStdDevPointCol;
+                lastMeanSubtractStdDevPointCol = meanSubtractStdDevPointCol;
+                lastPointRow = pointRow;
+                firstPoint = false;
+            }
+            // Draw median add StdDev line
+            abLine2D = new Line2D.Double(
+                    lastMeanAddStdDevPointCol,
+                    lastPointRow,
+                    meanAddStdDevPointCol,
+                    pointRow);
+            setPaint(Color.LIGHT_GRAY);
+            draw(abLine2D);
+            // Draw median subtract StdDev line
+            abLine2D = new Line2D.Double(
+                    lastMeanSubtractStdDevPointCol,
+                    lastPointRow,
+                    meanSubtractStdDevPointCol,
+                    pointRow);
+            setPaint(Color.LIGHT_GRAY);
+            draw(abLine2D);
+            // Draw median line
+            abLine2D = new Line2D.Double(
+                    lastMeanPointCol, lastPointRow, meanPointCol, pointRow);
+            setPaint(Color.DARK_GRAY);
+            draw(abLine2D);
+            // Set parameters
+            lastMeanPointCol = meanPointCol;
+            lastMeanAddStdDevPointCol = meanAddStdDevPointCol;
+            lastMeanSubtractStdDevPointCol = meanSubtractStdDevPointCol;
+            lastPointRow = pointRow;
+        }
+    }
+
+    public void drawLineChartUsingMinQ1MedianQ3Max() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 81 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 397-494 into a separate method.